src/kern/linux/helpers.cc

```diff
@@ -37,7 +37,22 @@
 
 #include "kern/linux/helpers.hh"
 
+/* The following pragmas are to fix a bug in GCC \w the CPP standard library,
+ * outlined here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562.
+ * This ignores the 'maybe-unitialized' warning when importing regex.
+ */
+
+// save diagnostic state
+#pragma GCC diagnostic push
+// ignore 'maybe-uniitialized' warnings
+#if !defined(__has_warning) || __has_warning("-Wmaybe-uninitialized")
```

Comment:
You may want to check for GCC version 12.1 here, too. No need to disable this warning for other GCCs.

I *think* `(__GNUC__ == 12 && __GNUC_MINOR__ >= 1)` will work, but tbh chatGPT gave me that code, so YMMV.